### PR TITLE
Catch exceptions thrown when retrieving response headers

### LINKED ISSUE #469: TraceUtilities crashes whole WebAPI application when ListAccessibleCustomers call fails
**Description**
When you turn on trace utilities, the lib will crash a WebAPI completely. Somehow this happens even in another WebAPI thread so try-catch doesn't help. Thus it also took quite some time to figure out this issue.

**Steps to Reproduce:**
Please provide a code snippet or steps to consistently reproduce the experienced behavior.
1. Create .NET Core WebApi project
2. Set it up with Google Ads API lib
3. Add this line to startup.cs: `TraceUtilities.Configure(TraceUtilities.DETAILED_REQUEST_LOGS_SOURCE, gadsLogPath, SourceLevels.Information);`
4. Execute: `customerService.ListAccessibleCustomers()` with a random string for a refresh token

**Expected behavior:**
Shouldn't crash the app, _try-catch_ should be able to handle the error.

**Client library version and API version:**
Client library version: latest
Google Ads API version: v11
.NET version: .NET 6.0
Operating system (Linux, Windows, ...) and version (if the bug is platform-specific): Windows

**Other**
I have a working prototype demonstrating this issue in case you need it.

### PR DESCRIPTION
**Pull request by Raibaz**
Fixes #469.

### CODE REVIEW

File: Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs [hunk 1]
```diff
@@ -82,11 +82,17 @@ internal LoggingHandler()
             where TRequest : class
             where TResponse : class
         {
-            // Generating log entry is expensive, so let's do that only if the log source
-            // has been configured to do so.
-
             RpcException exception = null;
```
Comment (AnashOommen) [inline Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs:85]: Should we put this behind a check for performance concerns?

var genSummaryLogs = TraceUtilities.ShouldGenerateSummaryRequestLogs();
var genDetailLogs = TraceUtilities.ShouldGenerateDetailedRequestLogs();
var genAnyLogs = genSummaryLogs || genDetailLogs;

if (!genAnyLogs) {
  return;
}

**Review by AnashOommen — CHANGES_REQUESTED** [review]
Left a comment.

File: Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs (hunk 1 above)
Comment (Raibaz) [reply Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs]: I don't think we are incurring a huge performance hit here, but i don't see any disadvantages in doing that check.

Added it.

**Review by AnashOommen — APPROVED** [review]

File: Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs [hunk 2]
```diff
@@ -99,24 +112,22 @@ internal LoggingHandler()
                     IsFailure = oldTask.IsFaulted,
                     Exception = exception,
                     CustomerId = GetCustomerId(request),
-                    ResponseHeaders = Merge(await call.ResponseHeadersAsync,
```
Comment (bitcras) [inline Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs:102]: Merge

File: Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs [hunk 3]
```diff
@@ -99,24 +112,22 @@ internal LoggingHandler()
...
-                                            call.GetTrailers()),
+                    ResponseHeaders = responseHeaders,
                     Response = (oldTask.IsFaulted) ? default : oldTask.Result,
                 };
 
                 WriteSummaryLogs(logEntry);
             }
 
-            if (TraceUtilities.ShouldGenerateDetailedRequestLogs())
+            if (generateDetailedLogs)
```
Comment (bitcras) [inline Google.Ads.GoogleAds.Core/src/Logging/LoggingHandler.cs:122]: Generate detailed logs

**Review by bitcras — APPROVED** [review]

**Pull request merged**